core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProfileManagerDAO.scala

```diff
@@ -0,0 +1,63 @@
+package org.broadinstitute.dsde.rawls.billing
+
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.rawls.config.MultiCloudWorkspaceConfig
+import org.broadinstitute.dsde.rawls.dataaccess.SamDAO
+import org.broadinstitute.dsde.rawls.model.{CreationStatuses, RawlsBillingProject, RawlsBillingProjectName, SamResourceAction, SamResourceTypeNames, UserInfo, WorkspaceAzureCloudContext}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+ * Facade over the billing profile manager service. This service will eventually by the source of truth for
+ * billing profiles in the Terra system. For now, we are using this to layer in "external" billing profiles
+ * for the purposes of testing Azure workspaces.
+ */
+class BillingProfileManagerDAO(samDAO: SamDAO, config: MultiCloudWorkspaceConfig) extends LazyLogging {
+
+  /**
+   * Fetches the billing profiles to which the user has access
+   */
+  def listBillingProfiles(userInfo: UserInfo)(implicit ec: ExecutionContext): Future[Seq[RawlsBillingProject]] = {
+    if (!config.multiCloudWorkspacesEnabled) {
+      return Future.successful(Seq())
+    }
+
+    val azureConfig = config.azureConfig match {
+      case None =>
+        logger.warn("Multicloud workspaces enabled but no azure config setup, returning empty list of billng profiles")
+        return Future.successful(Seq())
+      case Some(value) => value
+    }
+
+    // NB until the BPM is live, we are returning a hardcoded
+    // Azure billing profile, with access enforced by SAM
+    samDAO.userHasAction(
```

Comment:
There is an additional check that the UserService makes here https://github.com/broadinstitute/rawls/blob/ca50cb17a578da33abb4a1bb882fda940580ed56/core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala#L152 that should likely be factored into this class. It's fairly well entangled up with the rest of the code in that class that I decided to leave it there and defer until we are further down the road with the BPM integration.